### ISSUE TITLE
Incorrect error on internal hash reference

### ISSUE THREAD

**Issue by MrSeccubus** [gjtorikian/html-proofer#607]
I have the following HTML

```
<small><a href='#english'>English below</a></small>

<br>
<br>

<p><a href="https://www.linkedin.com/in/seccubus/">Frank Breedijk</a> is de initiatiefnemer en coordinator en pers woordvoerder van het Nederlands Security Meldpunt. In het dagelijks leven is hij de CISO van <a href="https://www.schubergphilis.com">Schuberg Philis</a>.</p>

<h3 id="english">English</h3>

<p><a href="https://www.linkedin.com/in/seccubus/">Frank Breedijk</a> is the initiator, coordinator and press contact of the Dutch Security Hotline. His day job is CISO of <a href="https://www.schubergphilis.com">Schuberg Philis</a>.</p>
```

Proofer produces the following error:
```
  *  linking to internal hash #english that does not exist (line 521)
     <a href="#english">English below</a>
```

I believe this to be a false positive.

**Comment by gjtorikian** [comment]
Can you share the command you are using to run your test? I can't reproduce this.

**Comment by MrSeccubus** [comment]
Yes, will do. Can upload the html code and command I use.

**Comment by MrSeccubus** [comment]
If I run the proof_html.sh script on the _site directory I get these errors:
```
  *  linking to internal hash #english that does not exist (line 521)
```
[2proofer.zip](https://github.com/gjtorikian/html-proofer/files/5671299/2proofer.zip)

Proofer command is `htmlproofer --allow-hash-href  --url-ignore="/www.linkedin.com/","/x1sec.com/" _site`

**Comment by gjtorikian** [comment]
Hmm. Could you confirm what version of html-proofer you are using?

**Comment by MrSeccubus** [comment]
When I do a `gem install html-roofer` in get
```
Fetching html-proofer-3.17.4.gem
```

**Comment by gjtorikian** [comment]
I simply cannot reproduce this. Irritating!

Your `2proofer.zip` file was lacking any HTML--would it be possible to add the contents of the failing file?

**Comment by chrisreddington** [comment]
Intriguingly, I seem to be using the same version of htmlproofer but I'm getting a different error [as shown here](https://github.com/chrisreddington/hugo-community/pull/44/checks?check_run_id=1539699705#step:7:6).

It seems to think internal links are external links. I'm happy to create a new issue if useful, but thought these two looked similar, so wanted to add here first.

**Comment by gjtorikian** [comment]
> It seems to think internal links are external links. I'm happy to create a new issue if useful, but thought these two looked similar, so wanted to add here first.

I've noticed you're using `htmlproofer 3.17.4`--could you try upgrading to the latest version and report back?

**Comment by chrisreddington** [comment]
> > It seems to think internal links are external links. I'm happy to create a new issue if useful, but thought these two looked similar, so wanted to add here first.
> 
> I've noticed you're using `htmlproofer 3.17.4`--could you try upgrading to the latest version and report back?

![image](https://user-images.githubusercontent.com/791642/102012270-539e8b80-3d41-11eb-947d-7511076c9a30.png)

Ok - Now using htmlproofer 3.18.0 in my local environment.

Looks like i'm getting some additional errors from running in my local environment, compared to a GitHub action runner - But, the issue is still there unfortunately.

**Comment by gjtorikian** [comment]
There seems to be some weird combination of HTML on the pages, but I was able to fix this in #610. I'll put out a patch release as soon as that's merged.
